packages/datadog-plugin-graphql/src/index.js

```diff
@@ -7,27 +7,30 @@ const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
 let tools
 
 function createWrapExecute (tracer, config, defaultFieldResolver) {
+  const wrappedDefaultFieldResolver = wrapResolve(defaultFieldResolver, tracer, config)
```

Comment:
This optimization is not needed since `wrapResolve` will be called later on anyway and already checks whether it's been patched.

Reply:
Won't we end up re-wrapping it on every execution, though?

Reply:
Not if it has a `_datadog_patched` property on the function which would always be the case if it went through `wrapResolve` at least once.

Reply:
But it won't – `wrapResolve` doesn't mutate its input arguments.

Reply:
It adds the `_datadog_patched` property at the end of the function, and checks for its presence at the beginning of the function to instantly return if present.